```bash
./homeassistant/components/sensor/serial.py:117:1: D202 No blank lines allowed after function docstring
./homeassistant/components/sensor/serial.py:117:1: D208 Docstring is over-indented
```

--- a/homeassistant/components/sensor/serial.py
+++ b/homeassistant/components/sensor/serial.py
@@ -114,9 +114,7 @@ class SerialSensor(Entity):
 
     @property
     def state_attributes(self):
-        """Return the attributes of the entity.
-
-           Provide the parsed JSON data (if any).
+        """Return the attributes of the entity (if any JSON present).
         """
 
         return self._attributes

--- a/homeassistant/components/sensor/serial.py
+++ b/homeassistant/components/sensor/serial.py
@@ -114,9 +114,7 @@ class SerialSensor(Entity):
 
     @property
     def state_attributes(self):
-        """Return the attributes of the entity (if any JSON present).
-        """
-
+        """Return the attributes of the entity (if any JSON present)."""
         return self._attributes
 
     @property

--- a/homeassistant/components/sensor/serial.py
+++ b/homeassistant/components/sensor/serial.py
@@ -84,11 +84,10 @@ class SerialSensor(Entity):
 
             try:
                 data = json.loads(line)
-            except json.JSONDecodeError:
-                _LOGGER.error("Invalid JSON data received: %s", data)
-
-            if isinstance(data, dict):
-                self._attributes = data
+                if isinstance(data, dict):
+                    self._attributes = data
+            except:
+                pass
 
             if self._template is not None:
                 line = self._template.async_render_with_possible_json_value(

